### PR TITLE
Support all visualization levels

### PR DESCRIPTION
**Pull request by wixette**
This pr is kind of large, mainly including the following logic:

* Localization support - Chinese and English assets/strings are supported with Unity localization package. See 423732f9af2da7a8f550870a512af7660940627c
* Models of the reference objects for visualizing numbers. See 40fa575f58b2bce5ce5c244b24bb6b26ecb3fd02 - Not that placeholder model is used when a model is not suitable to be opensourced.
* Font and Unit TMP (TextMeshPro) assets to change all UI texts to high-quality text. See a4087f6f6be51973110c667d1cf59f2683801a6c
* Code update to config and support the setup and the transition animations of all visualization levels. See 833745ad5465083325de15ad9a7b52aa523b5839

You can review each logic per its commit number. Thanks,

### CODE REVIEW

**Review by codingpotato — CHANGES_REQUESTED** [review]

File: Assets/Src/Scripts/SeedCalc/CuttingBoard.cs [hunk 1]
```diff
@@ -13,29 +13,51 @@
...
+using System.Collections.Generic;
 using UnityEngine;
 
 namespace SeedCalc {
   // The cutting board to show visualizable numbers and reference objects.
   public class CuttingBoard : MonoBehaviour {
-    public const int MinLevel = -11;
-    public const int MaxLevel = 9;
+    private class SlideAnimConfig {
+      internal GameObject Actor;
```
[the remark this answers is not in the document]
Comment (wixette) [reply Assets/Src/Scripts/SeedCalc/CuttingBoard.cs]: Done.

File: Assets/Src/Scripts/SeedCalc/CuttingBoard.cs [hunk 2]
```diff
@@ -44,43 +66,284 @@ public class CuttingBoard : MonoBehaviour {
...
-    // TODO: This is only for demonstrating the UI design. Remove it before production.
-    private IEnumerator AnimateAllLevels(float intervalInSeconds) {
-      Active = true;
-      for (int i = MinLevel; i <= MaxLevel; i++) {
-        Level = i;
-        yield return new WaitForSeconds(intervalInSeconds);
+    void Update() {
+      if (Input.GetMouseButtonDown(0)) {
+        var ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+        if (Physics.Raycast(ray, out RaycastHit hit, 100.0f)){
```
[the remark this answers is not in the document]
Comment (wixette) [reply Assets/Src/Scripts/SeedCalc/CuttingBoard.cs]: Done.

File: Assets/Src/Scripts/SeedCalc/LevelConfig.cs [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+
+namespace SeedCalc {
+  // Class to define the reference objects, the animations and the scale of a single level.
+  public class LevelConfig {
+    // Class to define a reference object in a level.
+    public class RefObjConfig {
+      // Every reference object must have an empty parent object as its container. The container
+      // object is used to position and scale the reference object.
+      public string ContainerName;
+      // The reference object itsself owns the animations that can be triggered by controllers. It
```
[the remark this answers is not in the document]
Comment (wixette) [reply Assets/Src/Scripts/SeedCalc/LevelConfig.cs]: Done.

File: Assets/Src/Scripts/SeedCalc/Nav.cs [hunk 4]
```diff
@@ -12,29 +12,40 @@
...
-using System.Collections;
+using TMPro;
 using UnityEngine;
 using UnityEngine.UI;
 
 namespace SeedCalc {
   // The nav bar to show the current level of the space grid.
   public class Nav : MonoBehaviour {
+    // The nav bar uses a different level system compared with the main cutting board, so that there
+    // is a flexibility to map more than one cutting board levels to the same nav bar level. The
```
[the remark this answers is not in the document]
Comment (wixette) [reply Assets/Src/Scripts/SeedCalc/Nav.cs]: Done.

File: Assets/Src/Scripts/SeedCalc/LevelConfig.cs [hunk 5]
```diff
@@ -0,0 +1,631 @@
...
+      // must also have a collider component so that it can receive users' click and touch events.
```
[the remark this answers is not in the document]
Comment (wixette) [reply Assets/Src/Scripts/SeedCalc/LevelConfig.cs:27]: It means Unity collider components, such as cube collider, sphere collider, mesh collider, etc. Without at least one collider component, a game object cannot be mouse-clicked or touched.

File: Assets/Src/Scripts/SeedCalc/LevelConfig.cs [hunk 6]
```diff
@@ -0,0 +1,631 @@
...
+    // one reference object.
+    public RefObjConfig[] RefObjs;
+
+    // The cutting board has a 4 rows by 6 columns grid, composed of 24 large square cells. Every
+    // large cell has a 10 x 10 grid inside. ScalePerLargeUnit defines the side length (in meters)
+    // of every large cell.
+    public double ScalePerLargeUnit;
+
+    // The formmatted value string of the scale marker length. Theoretically this value can be got
+    // from (ScalePerLargeUnit / 5.0), but we require a pre-formatted string here to provide a
```
[the remark this answers is not in the document]
Comment (wixette) [reply Assets/Src/Scripts/SeedCalc/LevelConfig.cs]: Done.

**Review by codingpotato — APPROVED** [review]
LGTM

**Pull request merged**